sql/src/main/java/io/crate/concurrent/CompletionListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.concurrent;
+
+import com.google.common.util.concurrent.FutureCallback;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public interface CompletionListener extends FutureCallback<CompletionState> {
+
+    CompletionListener NO_OP = new CompletionListener() {
+        @Override
+        public void onSuccess(@Nullable CompletionState state) {
+        }
+
+        @Override
+        public void onFailure(@Nonnull Throwable throwable) {
+        }
+    };
+
+    void onSuccess(@Nullable CompletionState state);
```

Comment:
can the `onSuccess` and `onFailure` definitions be ommitted as they're in the FutureCallback interface?